src/yunohost/backup.py

```diff
@@ -186,38 +188,47 @@ def _clean_tmp_dir(retcode=0):
 
         # Run apps backup scripts
         tmp_script = '/tmp/backup_' + str(timestamp)
-        for app_id in apps_filtered:
-            app_setting_path = '/etc/yunohost/apps/' + app_id
+        for app_instance_name in apps_filtered:
+            app_setting_path = '/etc/yunohost/apps/' + app_instance_name
 
             # Check if the app has a backup and restore script
             app_script = app_setting_path + '/scripts/backup'
             app_restore_script = app_setting_path + '/scripts/restore'
             if not os.path.isfile(app_script):
-                logger.warning(m18n.n('unbackup_app', app=app_id))
+                logger.warning(m18n.n('unbackup_app', app=app_instance_name))
                 continue
             elif not os.path.isfile(app_restore_script):
-                logger.warning(m18n.n('unrestore_app', app=app_id))
+                logger.warning(m18n.n('unrestore_app', app=app_instance_name))
 
-            tmp_app_dir = '{:s}/apps/{:s}'.format(tmp_dir, app_id)
+            tmp_app_dir = '{:s}/apps/{:s}'.format(tmp_dir, app_instance_name)
             tmp_app_bkp_dir = tmp_app_dir + '/backup'
-            logger.info(m18n.n('backup_running_app_script', app=app_id))
+            logger.info(m18n.n('backup_running_app_script', app=app_instance_name))
             try:
                 # Prepare backup directory for the app
                 filesystem.mkdir(tmp_app_bkp_dir, 0750, True, uid='admin')
                 shutil.copytree(app_setting_path, tmp_app_dir + '/settings')
 
                 # Copy app backup script in a temporary folder and execute it
                 subprocess.call(['install', '-Dm555', app_script, tmp_script])
-                hook_exec(tmp_script, args=[tmp_app_bkp_dir, app_id],
-                          raise_on_error=True, chdir=tmp_app_bkp_dir)
+
+                # Prepare env. var. to pass to script
+                env_dict = {}
+                app_id, app_instance_nb = _parse_app_instance_name(app_instance_name)
+                env_dict["YNH_APP_ID"] = app_id
+                env_dict["YNH_APP_INSTANCE_NAME"] = app_instance_name
+                env_dict["YNH_APP_INSTANCE_NUMBER"] = str(app_instance_nb)
```

Comment:
The `tmp_app_bkp_dir` is not set in the environment variables, I think we should add it if we want to go to the very end!